### PR TITLE
[`Flava`] Fix flava doc

### LINKED ISSUE #26078: broken FLAVA example code
### System Info

`transformers==4.33.1`

### Who can help?

@ArthurZucker @younesbelkada @amyeroberts 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Error is thrown when running the following [sample code from the FLAVA documentation](https://huggingface.co/docs/transformers/model_doc/flava#transformers.FlavaModel):
```
from PIL import Image
import requests
from transformers import AutoProcessor, FlavaModel

model = FlavaModel.from_pretrained("facebook/flava-full")
processor = AutoProcessor.from_pretrained("facebook/flava-full")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(text=["a photo of a cat"], images=image, return_tensors="pt", padding=True)

outputs = model(**inputs)
logits_per_image = outputs.contrastive_logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
```

Error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[9], line 1
----> 1 logits_per_image = outputs.contrastive_logits_per_image  # this is the image-text similarity score
      2 probs = logits_per_image.softmax(dim=1)

AttributeError: 'FlavaModelOutput' object has no attribute 'contrastive_logits_per_image'
```

### Expected behavior

Add `contrastive_logits_per_image` to output or fix example in documentation.

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes: https://github.com/huggingface/transformers/issues/26078

In fact the current flava docstring fails, running:

```bash
pytest --doctest-modules src/transformers/models/flava/modeling_flava.py::transformers.models.flava.modeling_flava.FlavaModel.forward
```

Leads to a failure. This PR fixes it

`contrastive_logits_per_image` cannot be retrieved from `FlavaModel` as detailed here: https://github.com/huggingface/transformers/issues/26078#issuecomment-1713369568 , therefore I have decided to modify the docstring to reflect the correct way of retrieving outputs from `FlavaModel`

### CODE REVIEW

File: src/transformers/models/flava/modeling_flava.py [hunk 1]
```diff
@@ -1371,8 +1371,10 @@ def forward(
         >>> inputs = processor(text=["a photo of a cat"], images=image, return_tensors="pt", padding=True)
 
         >>> outputs = model(**inputs)
-        >>> logits_per_image = outputs.contrastive_logits_per_image  # this is the image-text similarity score
-        >>> probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
+
+        >>> image_embeddings = outputs.image_embeddings
+        >>> text_embeddings = outputs.text_embeddings
+        >>> multimodal_embeddings = outputs.multimodal_embeddings
```
Comment (ydshieh) [inline src/transformers/models/flava/modeling_flava.py:1377]: maybe add the shape of those outputs ?

**Review by ydshieh — APPROVED** [review]
Thank you for working on this!

File: src/transformers/models/flava/modeling_flava.py [hunk 2]
```diff
@@ -1375,6 +1375,8 @@ def forward(
         >>> image_embeddings = outputs.image_embeddings
         >>> text_embeddings = outputs.text_embeddings
         >>> multimodal_embeddings = outputs.multimodal_embeddings
+        >>> print(image_embeddings.shape, text_embeddings.shape, multimodal_embeddings.shape)
+        torch.Size([1, 197, 768]) torch.Size([1, 7, 768]) torch.Size([1, 205, 768])
```
Comment (ydshieh) [inline src/transformers/models/flava/modeling_flava.py:1379]: I am not sure this works ..(?)

Comment (younesbelkada) [reply src/transformers/models/flava/modeling_flava.py:1379]: I ran the test with `pytest --doctest-modules src/transformers/models/flava/modeling_flava.py::FlavaModel.forward` and it passes, let me know if you had something else in mind!

Comment (ydshieh) [reply src/transformers/models/flava/modeling_flava.py:1379]: oh, I see, it is in a print statement so it works. Still not very prettry.

Usually we do sth like

```
>>> outputs.image_embeddings
torch.Size([1, 197, 768])

>>> outputs.text_embeddings
torch.Size([1, 7, 768])
```

Comment (younesbelkada) [reply src/transformers/models/flava/modeling_flava.py:1379]: I see ok ok, happy to change that as well

**Review by ArthurZucker — APPROVED** [review]
Thanks for fxing 💉

**Pull request merged**